components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -308,7 +308,12 @@ uint32_t AIChatTabHelper::GetMaxPageContentLength() {
   if (g_max_page_content_length_for_testing) {
     return *g_max_page_content_length_for_testing;
   }
-  return GetCurrentModel().max_page_content_length;
+  auto& model = GetCurrentModel();
+  if (model.option->is_leo_model()) {
+    return model.option->get_leo_model()->max_page_content_length;
+  } else {
+    return 0;
```

Comment:
@LorenzoMinto we need an optimal number for custom models here. this is in characers.

Reply:
Llama 3 has 8k tokens as max context length. Phi 3 Mini has 4k tokens (same as llama 2). We can use the second as an upper bound. 1 token ~ 4 characters as a rule of thumb. So max context length would be 16k characters. To give enough space for follow ups, we could have max page content length be 0.6*MAX_CONTENT_LENGTH, so 9.6k characters.

Reply:
changed